src/pygeoprocessing/geoprocessing.py

```diff
@@ -1487,6 +1524,10 @@ def zonal_statistics(
         aggregate_stats[unset_fid]['count'] = valid_unset_fid_block.size
         aggregate_stats[unset_fid]['nodata_count'] = numpy.count_nonzero(
             unset_fid_nodata_mask)
+        if include_value_counts:
+            aggregate_stats[unset_fid]['value_counts'].update(
```

Comment:
this might be worth a comment that `aggregate_stats[unset_fid]['value_counts']` is a `collections.Counter` and so `update` adds to it rather than overwrites.

Reply:
I agree, patched in 7dccc77